lstchain/calib/camera/calibrator.py

```diff
@@ -191,12 +190,16 @@ def _calibrate_dl1(self, event, telid):
         #      - gain selection before charge integration
 
 
-        no_gain_selection= np.zeros((waveforms.shape[0],waveforms.shape[1]), dtype=np.int)
+        no_gain_selection = np.zeros((waveforms.shape[0], waveforms.shape[1]), dtype=np.int)
 
-        charge, peak_time = self.image_extractor(waveforms, telid, no_gain_selection)
+        # the image extractor work per channel
+        charge0, peak_time0 = self.image_extractor(waveforms[0], telid, no_gain_selection[0])
```

Comment:
maybe clearer if you define `charge` and `peak_time` before:
```
charge = np.zeros(2)
peak_time = np.zeros(2)

charge[0], peak_time[0] = self.image_extractor(waveforms[0], telid, no_gain_selection[0])
charge[1], peak_time[1] = self.image_extractor(waveforms[1], telid, no_gain_selection[1])
```

Reply:
Hi @vuillaut, 
why do you propose to make the charge extractor to work per channel?

Reply:
I guess because this is how it was in ctapipe 0.7? ctapipe 0.8 only calculates one image, as it expects gain selection to be done before image extraction: https://github.com/cta-observatory/ctapipe/pull/1167

This is actually a change that should now (when completely switching to 0.8) be done in the event source, as ctapipe expects to start at the DL0 level now, where gain selection has already happened.

Reply:
Yes, as Max explains, the image extractor no longer accepts data cubes of dimensions > 2.
So... maybe there is no better solution if we want to keep analysing both channels for real data.

Reply:
@vuillaut, yes it does.  You do not need to separate HG and LG. Try for example:

```
from ctapipe.image.extractor import LocalPeakWindowSum
extractor = LocalPeakWindowSum(subarray=reader.subarray, window_width=11, window_shift=4)
selected_gain_channel = np.zeros((2,1855), dtype=np.int)
waveform=np.ones((2,1855,40))
waveform[0]=3
charge, pulse_time = extractor(waveform, 1, selected_gain_channel)

print(charge)
```

Which type of problems do to you have?